releng/jenkins/sign-and-deploy/Jenkinsfile

```diff
@@ -0,0 +1,272 @@
+            md5sum --binary $WORKSPACE/build-result/downloads/$ZIP_NAME > $WORKSPACE/build-result/downloads/$ZIP_NAME.md5
+          cd $WORKSPACE
+          
+        '''
+        } // END container
+        
+        // remote commands require the jnlp container
+        sshagent(['projects-storage.eclipse.org-bot-ssh']) {
+        container ('jnlp') {
+          sh '''
+          # TODO move function to external shell script
+          # $1 path to .properties file
+          # $2 property name
+          # @return property value
+          get_property () {
+            if [ ! -f "$1" ]; then
+              echo "File not found: $1" >&2
+              return -1
+            fi
+            VALUE=$(grep "^$2" $1 |awk -F= '{print $2}')
+            if [ -z "$VALUE" ]; then
+              echo "Could not get property value" >&2
+              return -2
+            fi
+            echo $VALUE
+          }
+          VERSION=$(get_property build-result/publisher.properties version)
+          BUILD_ID=$(get_property build-result/promote.properties build.id)
+          # BUILD_TYPE: N=Nightly, S=Stable, R=Release
+          BUILD_TYPE=$(echo $BUILD_ID |cut -c 1)
+          case "$BUILD_TYPE" in
+            N) ZIP_NAME=tmf-xtext-Update-$BUILD_ID.zip ;;
+            S|R) ZIP_NAME=tmf-xtext-Update-$VERSION.zip ;;
+          esac
+          
+          #
+          # STEP 3: Upload zip und .md5
+          #
+          TARGET_DROP_PATH=$DOWNLOAD_AREA/$VERSION/$BUILD_ID
+          # ensure target directory exists
+          ssh genie.xtext@projects-storage.eclipse.org "mkdir -p $TARGET_DROP_PATH"
+          # TODO: For stable/release builds fail when target location already exists
+          scp $WORKSPACE/build-result/downloads/$ZIP_NAME $WORKSPACE/build-result/downloads/$ZIP_NAME.md5 genie.xtext@projects-storage.eclipse.org:/$TARGET_DROP_PATH
+          
+          #
+          # STEP 4: Unpack zip to p2 repository location
+          #
+          case "$BUILD_TYPE" in
+            N) # Nightly site => clear content
+              REPOSITORY_PATH="/home/data/httpd/download.eclipse.org/modeling/tmf/xtext/updates/nightly"
+              ssh genie.xtext@projects-storage.eclipse.org "rm -rf $REPOSITORY_PATH/** && unzip -d $REPOSITORY_PATH $TARGET_DROP_PATH/$ZIP_NAME" ;;
+            S) # Stable
+              REPOSITORY_PATH="/home/data/httpd/download.eclipse.org/modeling/tmf/xtext/updates/milestones/$BUILD_ID"
+              ssh genie.xtext@projects-storage.eclipse.org "mkdir $REPOSITORY_PATH && unzip -d $REPOSITORY_PATH $TARGET_DROP_PATH/$ZIP_NAME" ;;
+            R) # Release
+              REPOSITORY_PATH="/home/data/httpd/download.eclipse.org/modeling/tmf/xtext/updates/releases/$VERSION"
+              ssh genie.xtext@projects-storage.eclipse.org "mkdir $REPOSITORY_PATH && unzip -d $REPOSITORY_PATH $TARGET_DROP_PATH/$ZIP_NAME" ;;
+          esac
+          '''
+        } // END sshagent
+        } // END container
```

Comment:
This is actually the other way around :relaxed:

Reply:
eagle eye!